Agents/AssetManagerAgent/AssetManagerAgent/src/test/java/uk/ac/cam/cares/jps/agent/assetmanager/AssetManagerAgentTest.java

```diff
@@ -0,0 +1,642 @@
+            String endpoint = set.getValue();
+            storeClientMap.put(ns, new RemoteStoreClient(endpoint, endpoint));
+        }
+        
+        // To create testAgent without an exception being thrown, SystemLambda is used to mock an environment variable
+        // To mock the environment variable, a try catch need to be used
+        /*ENV AGENTPROPERTIES "/root/agent.properties"
+        ENV ONTOMAPPROPERTIES "/root/ontologyMap.properties"
+        ENV TSSEARCH "/root/tsSearch.properties"
+        ENV FOLDERQR "/root/folderQR/"
+        ENV FOLDERMANUAL "/root/manuals/"*/
+        try {
+        	SystemLambda.withEnvironmentVariable("AGENTPROPERTIES", agentPropFile)
+                .and("ONTOMAPPROPERTIES", ontoMapPropFile)
+                .and("TSSEARCH", tsSearchPropFile)
+                .and("FOLDERQR", folderQRFilepath)
+                .and("FOLDERMANUAL", folderManualFilepath)
+                .execute(() -> {
+                agent = new AssetManagerAgent();
+        	 });
+        }
+        // There should not be any exception thrown as the agent is initiated correctly
+        catch (Exception e) {
+            System.out.println(e);
+            throw new IOException(e);
+        }
+
+        
+    }
+    
+    public static JSONObject parseJSONFile(String filename) throws JSONException, IOException {
+        String content = new String(Files.readAllBytes(Paths.get(filename)));
+        return new JSONObject(content);
+    }
+
+    @Before
+    public void createExampleData () throws IOException{
+        File file = new File("./src/test/java/uk/ac/cam/cares/jps/agent/devinst/exampleRequests/exampleRequest.json");
+        //for(String fileNames : file.list()) System.out.println(fileNames);
+        //System.out.println(file.exists());
+        JSONObject desc = parseJSONFile("./src/test/java/uk/ac/cam/cares/jps/agent/devinst/exampleRequests/exampleRequest.json");
+        //exampleRequest.put(KEY_CLIENTPROPERTY, KEY_CLIENTPROPERTY);
+        //exampleRequest.put(KEY_DESCRIPTOR, desc);
+
+    }
+
+    @Ignore("Test containers requires docker to function")
+    @Before
+    public void startContainers() throws IOException, URISyntaxException {
+		try {
+			// Start Blazegraph container
+			blazegraph.start();
+		} catch (Exception e) {
+			throw new JPSRuntimeException("Docker container startup failed. Please try running tests again");
+		}
+
+        //URI for blazegraph endpoint
+        URIBuilder builder = new URIBuilder().setScheme("http").setHost(blazegraph.getHost()).setPort(blazegraph.getFirstMappedPort()).setPath("/blazegraph/namespace");
+
+        // create a new namespace (endpoint) on blazegraph 
+        for (String namespace : namespaceList){
+            CloseableHttpClient httpclient = HttpClients.createDefault();
+            HttpPost postRequest = new HttpPost(builder.build());
+            postRequest.setEntity(new StringEntity(reqBody, ContentType.DEFAULT_TEXT));
```

Comment:
`NAMESPACE` placeholder in reqBody is not replaced with `namespace`, the post request creates only the `NAMESPACE` namespace repeatedly